llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12583,10 +12583,14 @@ struct AAAddressSpaceImpl : public AAAddressSpace {
         return true;
       if (isa<LoadInst>(Inst))
         MakeChange(Inst, const_cast<Use &>(U));
-      if (isa<StoreInst>(Inst)) {
+      else if (isa<StoreInst>(Inst)) {
         // We only make changes if the use is the pointer operand.
         if (U.getOperandNo() == 1)
           MakeChange(Inst, const_cast<Use &>(U));
+      } else if (isa<AtomicRMWInst>(Inst) || isa<AtomicCmpXchgInst>(Inst)) {
+        // We only make changes if the use is the pointer operand.
+        if (U.getOperandNo() == 0)
```

Comment:
Hardcoding the pointer operand index, use the AtoimcRMWInst::getPointerOperandIndex